extensions/wikia/Our404Handler/SpecialOur404Handler_body.php

```diff
@@ -67,4 +67,26 @@ public function doRender404() {
 		header( "X-Redirected-By: Our404Handler" );
 		header( sprintf( "Location: %s", $oTitle->getFullURL( $query ) ), true, 301 );
 	}
+
+	/**
+	 * This hook is called when about to force a redirect to a canonical URL
+	 * for a title when we have no other parameters on the URL.
+	 *
+	 * Return false when we want to prevent the redirect to the canonical URL
+	 * for Our404Handler special page (for non-English wikis)
```

Comment:
Oh, basically the Apache directs the request to Special:Our404Handler, which is the canonical URL and this hook is never called. The hook is called for cases when Special:Our404Handler is called, but the canonical URL for it is other like Spezial:Our404Handler (German naming of the namespace).